src/utils/calculateReClammInitAmounts.ts

```diff
@@ -0,0 +1,66 @@
+import {
+    type InputAmount,
+    type MinimalToken,
+    type CreatePoolReClammInput,
+    isSameAddress,
+    MathSol,
+} from 'src';
+import { computeInitialBalanceRatio } from '@balancer-labs/balancer-maths';
+
+export async function calculateReClammInitAmounts({
+    tokens,
+    givenAmountIn,
+    createPoolInput,
+}: {
+    tokens: MinimalToken[];
+    givenAmountIn: InputAmount;
+    createPoolInput: CreatePoolReClammInput;
+}): Promise<InputAmount[]> {
+    // sort the user provided tokens
+    const sortedTokens = tokens.sort((a, b) =>
+        a.address.localeCompare(b.address),
+    );
+
+    const { initialMinPrice, initialMaxPrice, initialTargetPrice } =
+        createPoolInput;
+
+    const givenTokenIndex = sortedTokens.findIndex((t) =>
+        isSameAddress(t.address, givenAmountIn.address),
+    );
+
+    const proportion = computeInitialBalanceRatio(
+        initialMinPrice,
+        initialMaxPrice,
+        initialTargetPrice,
+    );
+
+    let calculatedAmountIn: InputAmount;
+
+    // https://github.com/balancer/reclamm/blob/8207b33c1ab76de3c42b015bab5210a8436376de/test/reClammPool.test.ts#L120-L128
+    if (givenTokenIndex === 0) {
+        // if chosen token is first in sort order, we multiply
+        calculatedAmountIn = {
+            address: sortedTokens[1].address,
+            rawAmount: MathSol.mulDownFixed(
+                givenAmountIn.rawAmount,
```

Comment:
Haven't had time to confirm the maths but is using `rawAmount` correct here? @brunoguerios can maybe double check during his review.
Also I think its worth adding an integration test that includes a token of <18 decimals. WETH/USDC might be good as there are already some on Base you could use to sanity check values, etc.

Reply:
Since these are the amounts to be used in the init, I'd expect it to be rawAmount as well 👍 
But I agree that an integration test is good to rule out any unexpected issues with <18 decimals tokens.

Reply:
not sure what alternative to using `rawAmount` would be? :thinking:

Reply:
not sure what alternative to using `rawAmount` would be? :thinking:

Reply:
good thinking about adding test that uses token <18 decimals! :raised_hands:

Reply:
Nice catch.
I think this will be an issue when we have tokens with different decimals as it will keep the decimals of the given token.
I'd suggest casting to `TokenAmount` and using scaled18 for the math. Then pass rawAmount on the result. If should be able to handle scaling up/down for you.
Wondering how rates will affect this 🤔 
Definitely worth having an integration test to make sure it's working as expected 👍

Reply:
so tests definitely fail when one of the tokens is <18 decimals. 

struggling to understand if it is because of my updated `calculateReClammInitAmounts` logic with the scaling

or if it is because something needs to change with `initialMinPrice`, `initialMaxPrice`, and `initialTargetPrice` if one token has 18 decimals and the other has 6